kernel/dev/serial/rs232.cpp

```diff
@@ -12,17 +12,18 @@
  *
  */
 
+#include <stdarg.h>
 #include <arch/arch.hpp>
 #include <dev/serial/rs232.hpp>
 #include <mem/heap.hpp>
-#include <lib/mutex.hpp>
-#include <stdarg.h>
 #include <lib/stdio.hpp>
+#include <lib/mutex.hpp>
+#include <lib/string.hpp>
+#include <lib/ring_buffer.hpp>
 
-static ring_buff_t* read_buffer = NULL;
-uint8_t rs_232_line_index;
-uint16_t rs_232_port_base;
-mutex_t mutex_rs232("rs232");
+static uint16_t rs_232_port_base;
+static RingBuffer<uint8_t, 1024> ring;
```

Comment:
Why use `uint8_t` if we're enqueuing and dequeuing into `char`s?

Reply:
I feel like `uint8_t` is better for raw data, but there's no real reason. I can switch it so we can avoid a cast.

Reply:
Resolved in 04f2009.